src/kuduClient.ts

```diff
@@ -0,0 +1,113 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
My reasons for this:

- The kudu-api package doesn't have a TypeScript type def, it is easy to make a mistake calling it directly.
- The package also doesn't check for HTTP response code, caller is responsible for doing the check.
- Since there are more than one places calling this package (explorer, deployment), it benefits the callers if there is a typed wrapper doing some common house keeping tasks (ie check response code) for them.